src/resolvers/Mutation/createPost.ts

```diff
@@ -85,14 +87,44 @@ export const createPost: MutationResolvers["createPost"] = async (
     );
   }
 
-  // Creates new post.
+  if (args.data.pinned) {
+    // Check if the user has privileges to pin the post
+    const currentUserIsOrganizationAdmin = currentUser.adminFor.some(
+      (organizationId) => organizationId.toString() === args.data.organizationId
+    );
+
+    if (
+      !(currentUser!.userType === "SUPERADMIN") &&
+      !currentUserIsOrganizationAdmin
+    ) {
+      throw new errors.UnauthorizedError(
+        requestContext.translate(USER_NOT_AUTHORIZED_TO_PIN.message),
+        USER_NOT_AUTHORIZED_TO_PIN.code,
+        USER_NOT_AUTHORIZED_TO_PIN.param
+      );
+    }
+  }
+
+  // Creates new post
   const createdPost = await Post.create({
     ...args.data,
```

Comment:
What if the `pinned` variable is null, undefined? We should not allow those values in the database. Or even it they're not allowed, this should not result in resolver failing to create a post. Check how mongodb/mongoose deals with it and whether we need to transform the input before storing it.

Reply:
Used a check in the latest commit to only restrict it to true/false